packages/url/test/test.js

```diff
@@ -0,0 +1,162 @@
+      Promise.all([assertOutput(asserts.pngExport), assertExists(`output/${pnghash}`)])
+    ));
+
+  it('should use publicPath', () =>
+    run('./fixtures/png.js', { limit: 10, publicPath: '/foo/bar/' }).then(() =>
+      Promise.all([assertOutput(`var png = "/foo/bar/${pnghash}";\nexport default png;`)])
+    ));
+
+  it('should create a nested directory for the output, if required', () =>
+    run('./fixtures/png.js', { limit: 10, fileName: 'subdirectory/[hash][extname]' }).then(() =>
+      Promise.all([assertExists(`output/subdirectory/${pnghash}`)])
+    ));
+
+  it('should create a file with the name and extension of the file', () =>
+    run('./fixtures/png.js', { limit: 10, fileName: '[name][extname]' }).then(() =>
+      Promise.all([assertExists(`output/${pngname}`)])
+    ));
+
+  it('should create a file with the name, hash and extension of the file', () =>
+    run('./fixtures/png.js', { limit: 10, fileName: '[name]-[hash][extname]' }).then(() =>
+      Promise.all([assertExists(`output/png-${pnghash}`)])
+    ));
+
+  it('should prefix the file with the parent directory of the source file', () =>
+    run('./fixtures/png.js', { limit: 10, fileName: '[dirname][hash][extname]' }).then(() =>
+      Promise.all([assertExists(`output/fixtures/${pnghash}`)])
+    ));
+
+  it('should prefix the file with the parent directory of the source file, relative to the sourceDir option', () =>
+    run('./fixtures/png.js', {
+      limit: 10,
+      fileName: '[dirname][hash][extname]',
+      sourceDir: path.join(__dirname, '../')
+    }).then(() => Promise.all([assertExists(`output/test/fixtures/${pnghash}`)])));
+
+  it('should copy the file according to destDir option', () =>
+    run('./fixtures/png.js', {
+      limit: 10,
+      fileName: '[dirname][hash][extname]',
+      destDir: path.join(__dirname, 'output/dest')
+    }).then(() => Promise.all([assertExists(`output/dest/fixtures/${pnghash}`)])));
+
+  it('should create multiple modules and inline files', () =>
+    run(['./fixtures/svg.js', './fixtures/png.js'], {}, true).then(() =>
+      Promise.all([
+        assertOutput(asserts.pngInline, `${dir}/png.js`),
+        assertOutput(asserts.svgInline, `${dir}/svg.js`)
+      ])
+    ));
+
+  it('should create multiple modules and copy files', () =>
+    run(['./fixtures/svg.js', './fixtures/png.js'], { limit: 0 }, true).then(() =>
+      Promise.all([
+        assertOutput(asserts.pngExport, `${dir}/png.js`),
+        assertOutput(asserts.svgExport, `${dir}/svg.js`),
+        assertExists(`${dir}/${svghash}`),
+        assertExists(`${dir}/${pnghash}`)
+      ])
+    ));
+});
+
+function promise(fn, ...args) {
+  return new Promise((resolve, reject) =>
+    fn(...args, (err, res) => (err ? reject(err) : resolve(res)))
+  );
+}
+
+const DEFAULT_OPTIONS = {
+  publicPath: '',
+  emitFiles: true
+};
+
+function run(input, options, experimentalCodeSplitting = false) {
+  const writeOptions = {
```

Comment:
Just a heads up: We've now got a utilities that'll test a bundle as well as get the code for a bundle, so this function shouldn't be needed anymore. You can leverage `t.snapshot` in `ava` and you won't have to write the file to disk, which will make the `assertExists` function no longer needed as well.